### PR TITLE
add dafny libraries as submodule

### PR DESCRIPTION
**Pull request by yizhou7**
Following the [comment](https://github.com/dafny-lang/dafny/issues/1236#issuecomment-922506972) from @robin-aws , I am attempting to add the dafny library as a submodule located under `Test` directory.

### CODE REVIEW

**Comment by robin-aws** [comment]
This is passing but isn't actually cloning the submodule. :) Looks like we need to add `submodules: true` to the relevant `checkout` action.

**Review by robin-aws — CHANGES_REQUESTED** [review]
This is passing but isn't actually cloning the submodule. :) Looks like we need to add `submodules: true` to the relevant `checkout` action.

**Comment by yizhou7** [comment]
> This is passing but isn't actually cloning the submodule. :) Looks like we need to add `submodules: true` to the relevant `checkout` action.

@robin-aws Sorry, I am not super familiar with git  submodule. Is it possible to do something like `git clone --recurse-submodules` in or `git submodule update --init --recursive` in the CI? Or are you suggesting some changes on this repository itself?

**Comment by robin-aws** [comment]
No worries! You should just need to add `submodules: true` immediately below this line: https://github.com/dafny-lang/dafny/blob/master/.github/workflows/msbuild.yml#L114

**Comment by robin-aws** [comment]
Aha, the `lit.site.cfg` in the `libraries` module is not exactly the same as the one in `dafny`, in that they define `%dafny` differently: the former is assuming `dafny` is on the PATH, but the latter is running the Dafny CLI from source using `dotnet` instead.

I'll have to put some thought into how to resolve that.

File: .github/workflows/msbuild.yml [hunk 1]
```diff
@@ -112,6 +112,8 @@ jobs:
       uses: actions/checkout@v2
       with:
         path: dafny
+        submodules: true
+    - run: rm Test/libraries/lit.site.cfg
```
Comment (robin-aws) [inline .github/workflows/msbuild.yml:116]: HA, cute. :)

```suggestion
    - run: rm dafny/Test/libraries/lit.site.cfg
```

(you just haven't changed into the `dafny` directory yet at this point)

Worth adding a comment here too as to why you have to do this.

**Pull request merged**